plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,595 @@
+        requests = []
+
+        if want_afi.get('enabled') is True and have_afi.get('enabled') is True:
+            # only need to send a request if want from playbook is set to non default value and the setting currently configured is non default
+            requests.extend(self.get_delete_enabled_request(want_afi))
+        if want_afi.get('verify_mac') is False and have_afi.get('verify_mac') is False:
+            requests.extend(self.get_delete_verify_mac_request(want_afi))
+        if want_afi.get('vlans') is not None and have_afi.get('vlans') is not None and have_afi.get("vlans") != []:
+            # gathering list of vlans to be deleted. this section also handles cases where empty list of vlans is passed in
+            # which means delete all vlans
+            to_delete_vlans = have_afi["vlans"]
+            if len(want_afi["vlans"]) > 0:
+                to_delete_vlans = want_afi["vlans"]
+            # removing vlans that don't exist on device
+            for vlan in list(to_delete_vlans):
+                if vlan not in have_afi["vlans"]:
+                    to_delete_vlans.remove(vlan)
+            to_delete = {"afi": want_afi["afi"], "vlans": to_delete_vlans}
+            if len(to_delete["vlans"]):
+                requests.extend(self.get_delete_vlans_requests(to_delete))
+        if want_afi.get('trusted') is not None and have_afi.get('trusted') is not None and have_afi.get('trusted') != []:
+            # gathering list of interfaces to be deleted. this section also handles cases where empty list of interfaces is passed in which
+            #  means delete all trusted interfaces
+            to_delete_trusted = have_afi["trusted"]
+            if len(want_afi["trusted"]) > 0:
+                to_delete_trusted = want_afi["trusted"]
+            # removing interfaces that don't exist on device
+            for intf in list(to_delete_trusted):
+                if intf not in have_afi["trusted"]:
+                    to_delete_trusted.remove(intf)
+            to_delete = {"afi": want_afi["afi"], "trusted": to_delete_trusted}
+            if len(to_delete["trusted"]):
+                requests.extend(self.get_delete_trusted_requests(to_delete))
+        if want_afi.get('source_bindings') is not None and have_afi.get('source_bindings') is not None and have_afi.get('source_bindings') != []:
+            # gathering list of source bindings to be deleted. this section also handles cases where empty list of bindings is passed in which
+            #  means delete all trusted bindings
+            to_delete_bindings = have_afi["source_bindings"]
+            if len(want_afi["source_bindings"]) > 0:
+                to_delete_bindings = want_afi["source_bindings"]
+            # removing bindings that don't exist on device
+            for binding in list(to_delete_bindings):
+                if binding not in have_afi["source_bindings"]:
+                    to_delete_bindings.remove(binding)
```

Comment:
Same as trusted interfaces.
```suggestion
            to_delete_bindings = have_afi["source_bindings"]
            if len(want_afi["source_bindings"]) > 0:
                to_delete_bindings = want_afi["source_bindings"]
                # removing bindings that don't exist on device
                for binding in list(to_delete_bindings):
                    if binding not in have_afi["source_bindings"]:
                        to_delete_bindings.remove(binding)
```